cmd/notification.go

```diff
@@ -1317,8 +1317,13 @@ func (args eventArgs) ToEvent(escape bool) event.Event {
 	uniqueID := fmt.Sprintf("%X", eventTime.UnixNano())
 
 	respElements := map[string]string{
-		"x-amz-request-id":        args.RespElements["requestId"],
-		"x-minio-origin-endpoint": globalMinioEndpoint, // MinIO specific custom elements.
+		"x-amz-request-id": args.RespElements["requestId"],
+		"x-minio-origin-endpoint": func() string {
+			if globalMinioEndpoint != "" {
+				return globalMinioEndpoint
+			}
+			return getAPIEndpoints()[0]
```

Comment:
Wouldn't `globalLocalNodeName` work better for `x-minio-origin-endpoint`?

Reply:
globalLocalNodeName is just a hostname not an endpoint - endpoint is already constructed with `scheme, hostname and port`